tidy3d/components/spice/sources/dc.py

```diff
@@ -0,0 +1,48 @@
+"""
+Our DC sources ultimately need to follow this standard form if we want to enable full electrical integration.
+
+```
+11.3.2 .DC: DC Transfer Function
+
+General form:
+
+    .dc srcnam vstart vstop vincr [src2 start2 stop2 incr2]
+
+Examples:
+
+    .dc VIN 0.25 5.0 0.25
+    .dc VDS 0 10 .5 VGS 0 5 1
+    .dc VCE 0 10 .25 IB 0 10u 1u
+    .dc RLoad 1k 2k 100
+    .dc TEMP -15 75 5
+```
+
+"""
+
+from typing import Optional
+
+import pydantic.v1 as pd
+
+from tidy3d.components.base import Tidy3dBaseModel
+
+
+class AbstractStaticTransferSourceDC(Tidy3dBaseModel):
+    name: str
+    start: Optional[pd.PositiveFloat] = 0
+    stop: Optional[pd.PositiveFloat] | None = None
+    """
+    TODOMARC chat Either they define the stop or they define the step, how do we want to enforce, validator?
+    """
+
+    step: Optional[pd.PositiveFloat] = pd.Field(
+        1.0,
+        title="Bias step.",
+        description="By default, a solution is computed at 0 bias. "
+        "If a bias different than 0 is requested, DEVSIM will start at 0 and increase bias "
+        "at 'dV' intervals until the required bias is reached. ",
+    )
+    # TODO units
```

Comment:
I like how you organized this. It makes a lot of sense now that I see it! 
I do have a concern, though: the way you'd define a simulation here would differ from the way sources are defined in FDTD and HEAT, wouldn't it? The source would be inside a `TransferFunctionDC` and not at the simulation object level, right?

Reply:
So I believe as we enter more electronic simulations, the definition of what a `simulation` is will evolve since they can require a combination of DC stimuli and RF stimuli together. As such, we need a way to discretise the relevant simulation steps and combine analysis in flexible ways. This is different from the frequency domain FDTD method so I do think this makes sense to extend though.

On the way the sources get integrated, I am of the opinion in this case it does make sense because of the 1D parametric nature of this source that - still - might directly interact with other sources from other solves. Towards where we need to go I think this structure makes sense and can be integrated in the normal flow.

Reply:
I don't think the `simulation` object necessarily needs to evolve but maybe we need to think about having a `workflow` object with which we can define some sort of (loosely) coupled simulations (charge and heat can and should/will be fully coupled but I don't think we'll ever have fully coupled charge-RF or Charge-FDTD). What do you think? @momchil-flex @dbochkov-flexcompute

Reply:
Yes, sounds good! We could keep the current `***Simulation` definitions while combining them in some sort of super-class later on. We can revisit this but to be fair our initial discussions about having a single `Simulation`-like class that can contain all sorts of components (e.g. optical, heat, charge sources) and then trying to parse what to do internally based on that always sounded to me a bit too ambitious and potentially still not ideal even if we *did* manage to handle it well, because again it would be hard for the user to have a mental map of what exactly is happening (i.e. how our "translational layer" works). I think similarly to what we discussed about `MultiPhysicsMedium`, having the separate media that go into different solvers explicitly defined is a bit more cumbersome to initialize, but at least much more explicit from the user point of view. Similarly I would probably imagine us heading to a `MultiPhysicsSimulation` where simulations associated to various solvers are defined, as well as some specification about how they couple - even if this will not be the most concise way to do things. This could be the same as the `Workflow` object @marc-flex mentions, or more likely there could be a translation from a `MultiPhysicsSimulation` to a `Workflow` that explicitly shows the order of solves that will be done and how e.g. one monitor becomes a medium or a source in another simulation, so the user can exmaine it in advance.

Again, I'm sure we'll be iterating on this a lot but it's probably what I'm leaning towards after our recent discussions.